pyuvdata/mwa_corr_fits.py

```diff
@@ -0,0 +1,235 @@
+    corr_mapper={}
+    #iterate over the number of antennas (128) and x and y (2)
+    for inp1 in range(128):
+        for inp2 in range(128):
+            for p1 in range(2):
+                for p2 in range(2):
+                    index1=inp1*2+p1
+                    index2=inp2*2+p2
+                    corr_mapper[(pfb_outputs_to_inputs[index1],pfb_outputs_to_inputs[index2])]=[inp1,inp2,p1,p2]
+    return corr_mapper
+
+class MWACorrFITS(UVData):
+    """
+    Defines a MWA correlator fits-specific subclass of UVData for reading MWA 
+    correlator fits files. This class should not be interacted with directly, 
+    instead use the read_mwa_corr_fits method on the UVData class.
+    """
+    def read_mwa_corr_fits(self, filelist, run_check=True, check_extra=True, 
+                 run_check_acceptability=True):
+        """
+        Read in data from a list of MWA correlator fits files. 
+
+        Args:
+            filelist: The list of MWA correlator files to read from. Currently only
+                a single observation is supported. Must include one metafits file.
+            run_check: Option to check for the existence and proper shapes of
+                parameters after reading in the file. Default is True.
+            check_extra: Option to check optional parameters as well as required
+                ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                parameters after reading in the file. Default is True.
+        """
+        
+        metafits_file=None
+        file_dict={}
+        
+        for file in filelist:
+            if file.lower().endswith('.metafits'):
+                #check that have a metafits file
+                #TODO: ***fix this so that it allows for multiple metafits files!***
+                #for now, force only one metafits file
+                if metafits_file is not None:
+                    raise ValueError('multiple metafits files in filelist')
+                metafits_file = file
+            #organize data files
+            #TOOD: what to do with the metafits_ppds file?
+            elif file.lower().endswith('00.fits'):
+                if '00' not in file_dict.keys():
+                    file_dict['00']=[file]
+                else:
+                    file_dict['00'].append(file)
+            elif file.lower().endswith('01.fits'):
+                if '01' not in file_dict.keys():
+                    file_dict['01']=[file]
+                else:
+                    file_dict['01'].append(file)
+            elif file.lower().endswith('.mwaf'):
+                if 'flags' not in file_dict.keys():
+                    file_dict['flags']=[file]
+                else:
+                    file_dict['flags'].append(file)        
+            else:
+                continue
+        
+        #checks:
+        if len(file_dict['00'])!=len(file_dict['01']):
```

Comment:
Does this limit us to 2 files per observation? It's possible that can be larger...

Reply:
Each coarse channel typically has two files associated with it, so this is a check that one isn't missing. I do need to edit it so that someone could submit only the earlier or only the later subset of files.